mono/metadata/assembly.c

```diff
@@ -1251,13 +1251,25 @@ mono_stringify_assembly_name (MonoAssemblyName *aname)
 {
 	const char *quote = (aname->name && g_ascii_isspace (aname->name [0])) ? "\"" : "";
 
-	return g_strdup_printf (
-		"%s%s%s, Version=%d.%d.%d.%d, Culture=%s, PublicKeyToken=%s%s",
-		quote, aname->name, quote,
-		aname->major, aname->minor, aname->build, aname->revision,
-		aname->culture && *aname->culture? aname->culture: "neutral",
-		aname->public_key_token [0] ? (char *)aname->public_key_token : "null",
-		(aname->flags & ASSEMBLYREF_RETARGETABLE_FLAG) ? ", Retargetable=Yes" : "");
+	GString *str;
+	str = g_string_new (NULL);
+	g_string_append_printf (str, "%s%s%s", quote, aname->name, quote);
+	if (aname->has_version ==  TRUE)
+		g_string_append_printf (str,", Version=%d.%d.%d.%d", aname->major, aname->minor, aname->build, aname->revision);
+	if (aname->has_culture && aname->culture) {
+		if (*aname->culture)
+		g_string_append_printf (str, ", Culture=%s", aname->culture);
+		else
+		g_string_append_printf (str, ", Culture=%s", "neutral");
+	}
+	if (aname->has_public_key_token){
+		if (aname->public_key_token [0])
+			g_string_append_printf (str,", PublicKeyToken=%s%s", aname->public_key_token, (aname->flags & ASSEMBLYREF_RETARGETABLE_FLAG) ? ", Retargetable=Yes" : "");
+		else g_string_append_printf (str,", PublicKeyToken=%s%s", "null", (aname->flags & ASSEMBLYREF_RETARGETABLE_FLAG) ? ", Retargetable=Yes" : "");
+		}
```

Comment:
Could just `if (...retargetable...) { g_string_append_printf (str, ", Retargetable=Yes"); }`  after the public key token logic - then you wouldn't have to duplicate that `?:` expression.